views.py

```diff
@@ -57,28 +57,43 @@ def get_root(self, request):
         except:
             return Response("Filemanager instance with given name doesnot exists", status=status.HTTP_400_BAD_REQUEST)
         person = self.request.person
-        try:
-            folder = Folder.objects.get(
-                person=person, root=None, parent=None, filemanager=filemanager)
-        except Folder.DoesNotExist:
-            folder = Folder(person=person, root=None,
-                            parent=None, content_size=0, max_space=1024, filemanager=filemanager)
-            folder.save()
-
-            folder = Folder.objects.get(
-                person=person, root=None, parent=None)
-        except Folder.DoesNotExist:
-            folder = Folder(person=person, root=None,
-                            parent=None, content_size=0, max_space=1024)
-            folder.save()
-        except Folder.MultipleObjectsReturned:
-            return Response("more than one root folder found for same person", status=status.HTTP_409_CONFLICT)
+        for filemanager in FileManager.objects.all():
+            try:
+                folder = Folder.objects.get(
```

Comment:
make it like a decorator or permission so that it can be reused in other views also.
it is not a logical solution to put same code in different views